### ISSUE TITLE
[Minor Improvement] Indicator loadout

### ISSUE THREAD

**Issue by Gab0** [askmike/gekko#1293]
[ ] bug report
[ ] feature request
[x] question about the decisions made in the repository

To create a new indicator, one should add it to the list `Indicators` at `tradingAdivisor/baseTradingMethod.js`
There is a specific reason for this? We could replace that with a search on indicators folder:
```
var indicatorFiles = fs.readdirSync(indicatorsPath);
var Indicators = {};

indicatorFiles.forEach(function(indicator){
    var indicatorName = indicator.split('.')[0];
    if (indicatorName[0] != '_') //ignore those starting with '_', so we have some control on which aren't loaded;
        Indicators[indicatorName] = require(indicatorsPath + indicator);
});
```
Then fix that line where `factory` attribute is required
```
 this.indicators[name] = new Indicators[type](parameters);
```
And set the input type (`candle` or `price`) on each indicator init function.

That would ease the process of creating and renaming indicators..

**Comment by generalectric** [comment]
I dont hate your idea.... but I disagree with it. This makes writing indicators easier being that you dont have to mod basetradingmethod and copy paste then edit a couple words.... but this creates more work for end users.. who might not know if an indicator is price or candle data... and I think there are other things more important than making everyone edit all their existing strats to conform with this?

**Comment by Gab0** [comment]
I see your point.. but man, it will be even easier to see that determined indicator takes one or other input type, because it will be declared on the indicator file itself:

>indicators/DEMA.js [config function]
```
var Indicator = function(config) {
  this.input = 'price'	
  this.result = false;
  this.short = new EMA(config.short);
  this.long = new EMA(config.long);
}
```
basetradingmethod.js and indicator files will be the only affected, no edit on strategies will be required. 
Thats for a very small improvement, for a bit of ease on indicator management. buy hey, its for free x]

**Comment by askmike** [comment]
I think this is a great idea, and I think it's a nice start to some next steps, that are:

Allow for anyone to easily add indicators, for this we would need to apply your proposed fix and create some documentation on how indicators are supposed to behave and voila :)

**Comment by askmike** [comment]
@generalectric 

> but this creates more work for end users.. who might not know if an indicator is price or candle data... and I think there are other things more important than making everyone edit all their existing strats to conform with this?

It is confusing that some indicators are always supposed to calculated over a specific value and others are not (like EMA). We should just make a nice API for this and explain it properly.

That said: I want to move Gekko towards a platform where people with different levels of programming experience all can do fun things with it, the UI is already a step towards it (very low boundry entry point). For example, this is from "beginner" to "expert":

1. install Gekko and use the UI to backtest/play around with default strategies (MACD, RSI, etc) and tweak the parameters.
2. use gekko over the commandline.
3. tweak existing strategies (intro javascript).
4. create your own strategies from scratch.
5. add indicitors.
6. add exchanges / plugins.
7. work on gekko's core.

(This is not a fixed list, as frontend developers familar with Vue should have an easy time picking up working on the UI, etc).

**Comment by askmike** [comment]
This has since been fixed in #1301, will hit stable in the next release!
